### PR TITLE
rust cork protos

### PR DESCRIPTION
**Pull request by cbrit**
Generated the rust proto types for the cork module and refactor the module somm_proto exports.

### CODE REVIEW

File: somm_proto/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 pub use cosmos_sdk_proto;
-pub mod somm {
+pub mod allocation {
```
Comment (EricBolten) [inline somm_proto/src/lib.rs:2]: Note: changing this might require updates in steward.

File: prost_build/src/main.rs [hunk 2]
```diff
@@ -21,15 +21,15 @@ const EXCLUDED_PROTO_PACKAGES: &[&str] = &["gogoproto", "google", "tendermint",
 const COSMOS_SDK_PROTO_REGEX: &str = "(super::)+cosmos";
 
 /// the output directory
+const TMP_PATH: &str = "/tmp/sommelier";
```
Comment (EricBolten) [inline prost_build/src/main.rs:24]: Why move to a hardcoded path rather than using the environment's tempdir?

**Review by EricBolten — COMMENTED** [review]
A question and a note.

File: somm_proto/src/lib.rs (hunk 1 above)
Comment (cbrit) [reply somm_proto/src/lib.rs:2]: It does, I've made them in my cork PR which depends on this change

File: prost_build/src/main.rs (hunk 2 above)
Comment (cbrit) [reply prost_build/src/main.rs:24]: It was causing weird file permissions errors and this is the way other proto builds do it in our repos

**Review by EricBolten — APPROVED** [review]
LGTM

**Pull request merged**